homeassistant/components/github/__init__.py

```diff
@@ -64,12 +65,32 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
         hass.data[DOMAIN][repository] = coordinators
 
+    await async_cleanup_device_registry(hass=hass, entry=entry)
+
     hass.config_entries.async_setup_platforms(entry, PLATFORMS)
     entry.async_on_unload(entry.add_update_listener(async_reload_entry))
 
     return True
 
 
+async def async_cleanup_device_registry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+) -> None:
+    """Remove entries form device registry if we no longer track the repository."""
+    device_registry = dr.async_get(hass)
+    for dev_id, device_entry in list(device_registry.devices.items()):
```

Comment:
The correct approach is to use `dr.async_entries_for_config_entry` with `device_registry.async_update_device(entry, remove_config_entry_id=entry.id)`

The device will be removed when no config entries are linked to it.

Reply:
Are you sure this is it?
I can't find a single place it's used like that?